shared/p2p/adapter/tracer/tracer.go

```diff
@@ -0,0 +1,54 @@
+package tracer
+
+import (
+	"context"
+	"errors"
+
+	"github.com/prysmaticlabs/prysm/shared/p2p"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+)
+
+const (
+	defaultSamplingFraction = 0.25
+)
+
+var log = logrus.WithField("prefix", "tracer")
+
+func New(name, endpoint string, disable bool) (p2p.Adapter, error) {
+	if disable {
+		trace.ApplyConfig(trace.Config{DefaultSampler: trace.NeverSample()})
+		return adapter, nil
+	}
+
+	if name == "" {
+		return nil, errors.New("tracing service name cannot be empty")
+	}
+
+	// TODO: make sampling fraction configurable?
```

Comment:
Thoughts on this? Currently hard-coded to sampling 25% of requests.

Reply:
That'd be awesome. Feel free to open an issue and change it to TODO(#Issue number)

Reply:
Done, made this configurable.

Reply:
Please format as TODO(#issue number)